### ISSUE TITLE
1.5.9 breaks `gitlabCommitStatus(name)` usage

### ISSUE THREAD

**Issue by crussell52** [jenkinsci/gitlab-plugin#813]
Edit: Removed "Before submitting..." paragraph
### Context
- **Gitlab plugin version**: 1.5.9
- **Gitlab version**: 9.5.4
- **Jenkins version**: 2.134
- **Job type**: Pipeline

### Logs & Traces
`Jenkinsfile`
```
gitlabBuilds(builds: ['Compile', 'Test', 'Package']) {
  node {
    ...
    stage("Compile") {
      gitlabCommitStatus("Compile") {
      ...
      }
  ...
  }
  ...
}
```

`Console Output`
```
java.lang.IllegalArgumentException: Expected named arguments but got Compile
	at org.jenkinsci.plugins.workflow.cps.DSL.singleParam(DSL.java:542)
	at org.jenkinsci.plugins.workflow.cps.DSL.parseArgs(DSL.java:530)
	at org.jenkinsci.plugins.workflow.cps.DSL.parseArgs(DSL.java:470)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeStep(DSL.java:183)
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:153)
       ...
```

### Problem description

_Describe your problem in a meaningful way_:

Builds fail with above stack under 1.5.9. 
Rolling back to 1.5.8 fixes the problem

Might be related to: https://github.com/jenkinsci/gitlab-plugin/commit/799e43326530eb7f8cca569774ea2a4504c72a3c

**Comment by omehegan** [comment]
*sigh*

There is clearly a gap in testing here...

@stchar can you look at this?

**Comment by omehegan** [comment]
@stchar ping?

**Comment by stchar** [comment]
@omehegan, @crussell52, sorry missed the message notification. 
Yep, you right lack of testing. I will fix it asap. Then I will try to add a regression.

**Comment by omehegan** [comment]
@stchar any update on this?

**Comment by karlskewes** [comment]
Thanks for lodging issue, just found this and updated Jenkinsfile's to include `name: ` before the Gitlab stage name. 
eg:
before: `      gitlabCommitStatus("Compile") {`
after: `      gitlabCommitStatus(name: "Compile") {`

**Comment by stchar** [comment]
@omehegan, @crussell52 
I'm not familiar with stapler. It's definitely  issue with usage of `@DataBoundConstructor` decorator.
Looks like it's behavior depends on jenkins version, hope it doesn't depend on any of pipeline plugin version.

Anyway I'm trying to restore the original constructor which had been here before I submitted my PR https://github.com/jenkinsci/gitlab-plugin/pull/747
```
    @DataBoundConstructor
    public GitLabCommitStatusStep(String name){
        this.name = StringUtils.isEmpty(name) ? null : name;
    }
```

I'm also setting up different environments to verify the issue on a vary of Jenkins core and pipeline plugin versions.

**Comment by omehegan** [comment]
@stchar thanks for working on it, if you need help with that area of the code (Stapler etc.) I would suggest posting on the https://groups.google.com/forum/#!forum/jenkinsci-dev mailing list.

**Comment by killianbrackey** [comment]
Hey @stchar! Any progress on this? I am happy to help or test any items that you have started in on or picking up from where you left off.

**Comment by stchar** [comment]
@killianbrackey, I think the fix is ready, 
I tested it with lts versions 2.107,  2.121.2
Only 2.143 or latest left as mentioned in the issue

You can build the plugin with my fix, or take one from the verification job:   
https://ci.jenkins.io/job/Plugins/job/gitlab-plugin/job/PR-823/9/artifact/target/gitlab-plugin-1.5.11-SNAPSHOT.hpi

**Comment by crussell52** [comment]
@stchar That's great news! Is there anything we can do to help get it published?

**Comment by kappacrafter** [comment]
@stchar, thank you for snapshot, works great! Waiting for new release.

**Comment by stchar** [comment]
@kappacrafter, thanks for assist, and sorry for a delayed fix.

**Comment by crussell52** [comment]
@omehegan @stchar - Any ETA on a release for this?

**Comment by omehegan** [comment]
@crussell52 sorry for the delay, I have not had time to work on the plugin for awhile. I'll test and merge this now if it's all good.

**Comment by omehegan** [comment]
Released in 1.5.11.

**Comment by dtronche** [comment]
I know this is a closed issue but by any chance could this explain why there is a regression on [795](https://github.com/jenkinsci/gitlab-plugin/issues/795) behaviour. Because I have the following error message:
```
WorkflowScript: 6: Missing required parameter: "name" @ line 6, column 17.
      gitlabCommitStatus {
```
I could understand name becomes a compulsary parameter but then the documentation should be updated as it clearly says it is optional
